src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -264,13 +229,13 @@ public async Task<bool> Initialize(
 
         if (RuntimeConfigLoader.TryParseConfig(jsonConfig, out RuntimeConfig? runtimeConfig, replaceEnvVar: replaceEnvVar, replacementFailureMode: replacementFailureMode))
         {
-            _runtimeConfig = runtimeConfig.DataSource.DatabaseType switch
+            ConfigLoader.RuntimeConfig = runtimeConfig.DataSource.DatabaseType switch
```

Comment:
are we ok setting this directly in a different class? should we instead have a private setter?

Reply:
I created an issue for more refactor to eliminate this being public, for now we need to be able to change the RuntimeConfig from the provider since we have parse functions that are not able to save the state of the RuntimeConfig directly inside of the loader. This is mainly because the hosted scenario uses static functions and modifies the runtimeconfig outside the loader through the out param.